File: src/main/resources/static/js/service/taskService.js
'use strict';

App.factory('TaskService', ['$http', '$q', function($http, $q){

    var HOST = "http://localhost:8080/tasks";

    return {

        loadAllTasks: function(page, count, userId) {
            return $http.get(HOST + '?page=' + page + '&count=' + count + '&userId=' + userId)
                .then(
                    function(response){
                        return response.data;
                    },
                    function(errResponse){
                        console.error('Error while load tasks');
                        return $q.reject(errResponse);
                    }
                );
        },

        createTask: function(task){
            return $http.post(HOST + '/add', task)
                .then(
                    function(response){
                        return response.data;
                    },
                    function(errResponse){
                        console.error('Error while creating task');
                        return $q.reject(errResponse);
                    }
                );
        },

        stopTask: function(taskId, userId){
            return $http.get(HOST + '/stop?taskId=' + taskId + '&userId=' + userId)
                .then(
                    function(response){
                        return response.data;
                    },
                    function(errResponse){
                        console.error('Error while updating task');
                        return $q.reject(errResponse);
                    }
                );
        },

        deleteTask: function(taskId, userId){
            return $http.delete(HOST +'?taskId=' + taskId + '&userId=' + userId)
                .then(
                    function(response){
                        return response.data;
                    },
                    function(errResponse){
                        console.error('Error while deleting task');
                        return $q.reject(errResponse);
                    }
                );
        }

    };

}]);

File: src/main/java/com/taskmanager/service/api/TaskService.java
package com.taskmanager.service.api;

import com.taskmanager.model.Task;
import org.springframework.data.domain.Page;

/**
 * Service for tasks
 */
public interface TaskService {

    /**
     * Add new task
     * @param task {@link Task}
     */
    void add(Task task);

    /**
     * Stop task by taskId and userId
     * @param taskId taskId
     * @param userId userId
     */
    void stop(String taskId, String userId);

    /**
     * Remove task by taskId and userId
     * @param taskId taskId
     * @param userId userId
     */
    void remove(String taskId, String userId);

    /**
     * Load tasks on page
     * @param page number of page
     * @param count number on page
     * @param userId userId
     * @return {@link Page} of {@link Task}
     */
    Page<Task> load(int page, int count, String userId);
}
